src/psp.ts

```diff
@@ -0,0 +1,664 @@
+            // **************************************************************************
+            node.logger.debug('Fetching gas price...', subscriptionIdLogOptions);
+
+            const [gasPriceLogs, gasTarget] = await node.evm.getGasPrice({
+              provider,
+              chainOptions: chain.options,
+            });
+            if (!isEmpty(gasPriceLogs)) {
+              node.logger.logPending(gasPriceLogs, subscriptionIdLogOptions);
+            }
+            if (!gasTarget) {
+              node.logger.warn('Failed to fetch gas price. Skipping update...', subscriptionIdLogOptions);
+              continue;
+            }
+
+            // **************************************************************************
+            // Compute signature
+            // **************************************************************************
+            node.logger.debug('Signing fulfill message...', subscriptionIdLogOptions);
+
+            const timestamp = Math.floor(Date.now() / 1000);
+
+            const signature = await airnodeWallet.signMessage(
+              ethers.utils.arrayify(
+                ethers.utils.keccak256(
+                  ethers.utils.solidityPack(
+                    ['bytes32', 'uint256', 'address'],
+                    [subscriptionId, timestamp, sponsorWallet.address]
+                  )
+                )
+              )
+            );
+
+            // **************************************************************************
+            // Update beacon
+            // **************************************************************************
+            node.logger.debug('Fulfilling subscription...', subscriptionIdLogOptions);
+
+            let fulfillFunction: ethers.utils.FunctionFragment;
+            try {
+              fulfillFunction = dapiServer.interface.getFunction(fulfillFunctionId);
+            } catch (error) {
+              node.logger.error('Failed to get fulfill function', {
+                ...subscriptionIdLogOptions,
+                error: error as any,
+              });
+              continue;
+            }
+            const currentNonce = nonce;
+            const [errfulfillFunction, tx] = await retryGo<ethers.ContractTransaction>(() =>
+              dapiServer
+                .connect(sponsorWallet)
+                .functions[fulfillFunction.name](
+                  subscriptionId,
+                  airnodeAddress,
+                  relayer,
+                  sponsor,
+                  timestamp,
+                  encodedFulfillmentData,
+                  signature,
+                  {
+                    gasLimit: GAS_LIMIT,
+                    ...gasTarget,
+                    nonce: nonce++,
+                  }
+                )
+            );
+            if (errfulfillFunction) {
+              node.logger.error(
+                `failed to submit transaction using wallet ${sponsorWallet.address} with nonce ${currentNonce}`,
```

Comment:
Uses the incremented nonce

Reply:
This might be a bit misleading but in reality the fulfill tx is submitted with the value of `currentNonce` because 'nonce++' is incrementing the value after assignment. Maybe I should separate this into a new line to make it more explicit and easier to read.

Reply:
Yeah I misread his. However, it feels like this won't work when you make multiple transactions and increment `nonce` multiple times.